tests/unit_tests/cpu/geometry/masks/rectangle2D.cpp

```diff
@@ -59,6 +60,36 @@ GTEST_TEST(detray_masks, rectangle2D) {
     ASSERT_NEAR(centroid[2], 0.f, tol);
 }
 
+/// This tests the inside/outside method of the mask
+GTEST_TEST(detray_masks, rectangle2D_monte_carlo) {
+
+    struct mask_check {
+        bool operator()(const test::point3 &p, const mask<rectangle2D> &r,
+                        const test::transform3 &trf, const scalar t) {
+
+            const test::point3 loc_p{r.to_local_frame(trf, p)};
+            return r.is_inside(loc_p, t) == intersection::status::e_inside;
+        }
+    };
+
+    constexpr mask<rectangle2D> r{0u, 3.f, 4.f};
+
+    constexpr scalar t{0.f};
+    const test::transform3 trf{};
+    constexpr scalar world{10.f * unit<scalar>::mm};
+    constexpr std::size_t n_steps{500u};
+
+    std::vector<test::point3> points =
+        test::generate_regular_points(world, n_steps);
```

Comment:
Why don't you test it with `generate_random_points`?

Reply:
Do you think it will make a significant difference in the test? Otherwise it'll probably just make the CI run longer

Reply:
A seeded random number generator just spaces the points irregularly, instead of reguarly. I'll rename it then